src/cr/cube/matrix/subtotals.py

```diff
@@ -209,6 +209,13 @@ class SumDiffSubtotals(SumSubtotals):
 
     def _intersection(self, row_subtotal, column_subtotal):
         """Sum and Diff for this row/column subtotal intersection."""
+        # --- Intersections of subtotal differences are undefined ---
+        if (
+            len(column_subtotal.subtrahend_idxs) > 0
+            and len(row_subtotal.subtrahend_idxs) > 0
+        ):
+            return np.nan
```

Comment:
This doesn't work because unweighted counts are currently stored as integers which don't have a `nan`. Fix should probably be to make unweighted counts as floats

Reply:
Next commit is a way to fix, but I'm not sure it's the correct way.

Reply:
Yeah, it's a good question. There are two approaches I can think of:

1. Adopt the rule that _all_ numeric measure values (but not idxs like `.inserted_column_idxs`) are float, and remove that broadly docstring-documented and otherwise implemented distinction between int and float values. I kind of think this is the right approach, but it would entail a lot of small changes, including right down to the cube-measure level where we would "cast" the ints we get for unweighted counts to floats.

2. Just remove the `dtype` setting here in subtotals and let numpy determine what type to make things and some are ints and some are floats and we don't pay too much attention. This is easy, but it feels sloppy and the kind of thing that would come to bite us later. As a general principle, we'd like to have a consistent interface, which includes consistent value types.

Let me know what you think and maybe we can discuss at stand-up.

Reply:
Okay, I made the changes discussed (`_UnweightedCountMeasure._flat_values` now converts to ndarrays of floats). I also took out the `_dtype` methods in Subtotals in `matrix.insertions.py` because it didn't seem useful now that everything should be a float.

I have not updated docstrings. I'm a little worried about making that many changes while you, Ernesto and I all need to rebase on each other. Do you think I need to do that now?

Reply:
Well, we probably should I suppose. My branch is on master now, so the bulk is already handled, and the bits that Ernesto is doing is primarily adds rather than changes, so I don't think the merge conflicts are going to be too big for him. I'm pretty sure yours will go first, because he's got a lot of changes to do on his branch and I expect it will be Monday or so before he's ready to push.